src/neo/SmartContract/ApplicationEngine.cs

```diff
@@ -224,10 +224,11 @@ private ExecutionContext CallContractInternal(ContractState contract, ContractMe
             ExecutionContextState state = CurrentContext.GetState<ExecutionContextState>();
             UInt160 callingScriptHash = state.ScriptHash;
             CallFlags callingFlags = state.CallFlags;
+            const CallFlags unlocks = CallFlags.AllowExceptions;
 
             if (args.Count != method.Parameters.Length) throw new InvalidOperationException($"Method {method} Expects {method.Parameters.Length} Arguments But Receives {args.Count} Arguments");
             if (hasReturnValue ^ (method.ReturnType != ContractParameterType.Void)) throw new InvalidOperationException("The return value type does not match.");
-            ExecutionContext context_new = LoadContract(contract, method, flags & callingFlags);
+            ExecutionContext context_new = LoadContract(contract, method, flags & (callingFlags | unlocks));
```

Comment:
But it's the caller who sets the `flags` and it's the caller that will try to attack some benign buggy contract. So as an attacker of course I will set `AllowExceptions` before calling the contract that is expected to perform some interesting state changes before throwing an exception.

Reply:
A(attacker) -> B(modify_stroage) -> C(throw)

I think it's B to set `callingFlags`.

Reply:
How about `A (attacker) -> B (modify_storage) -> B (throw)`?

Reply:
> 

Should we modify NEO and GAS's callflag?

Reply:
> How about A (attacker) -> B (modify_storage) -> B (throw)?

Then B should use `ABORT` instead of `THROW`.

Reply:
I thought again, I would close this pr first and create a new one in devpack.

Reply:
@roman-khimov Please check neo-project/neo-devpack-dotnet#740